web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java

```diff
@@ -0,0 +1,208 @@
+                "\"target\":[\"dimension\",[\"template-tag\",\"replicate_id\"]]," +
+                "\"value\":[\"" + participantLabels[1] + "\"]}";
+
+        HashMap<String, String> params = new HashMap<String, String>();
+        params.put("origin", origin);
+        params.put("replica", replica);
+
+        String url = insightURL + "/api/card/" + reportID + "/query/csv?parameters=[{origin},{replica}]";
+        String response = doPost(url, headers, null, params);
+
+        if (response != null) {
+            Study site = studyDao.findByOcOID(studyOid);
+            if (site != null) {
+                Study parentStd = site.getStudy();
+                if (parentStd != null) {
+                    utilService.setSchemaFromStudyOid(studyOid);
+                    UserAccountBean userAccountBean = utilService.getUserAccountFromRequest(request);
+                    UserAccount userAccount = userAccountDao.findById(userAccountBean.getId());
+
+                    String fileName = saveToFile(response.toString(), participantLabel);
+                    if (fileName != null) {
+                        JobDetail jobDetail= userService.persistJobCreated(parentStd, site, userAccount, JobType.INSIGHT_REPORT, fileName);
+                        userService.persistJobCompleted(jobDetail, fileName);
+                    }
+                }
+            }
+        }
+
+        return new ResponseEntity(HttpStatus.OK);
+    }
+
+    public String[] getLabels(String label) {
+        // original label/id not contain replicaSubString
+        // replica label/id contain replicaSubString
+        String[] labels = new String[2];
+        if (label.indexOf(replicaSubstring) > -1) {
+            labels[0] = label.split(replicaSubstring)[0];
+            labels[1] = label;
+        } else {
+            labels[0] = label;
+            labels[1] = label + replicaSubstring;
+        }
+        return labels;
+    }
+
+    public String doPost(String api, HashMap<String, String> mapHeaders,
+                         String body, HashMap<String, String> queries) {
+        try {
+            HttpHeaders headers = new HttpHeaders();
+            for (String i : mapHeaders.keySet()) {
+                headers.add(i, mapHeaders.get(i).toString());
+            }
+
+            HttpEntity<String> request;
+            if (body != null) {
+                request = new HttpEntity<>(body, headers);
+            } else {
+                request = new HttpEntity<String>(headers);
+            }
+            RestTemplate rest = new RestTemplate();
+            ResponseEntity<String> response;
+            if (queries != null) {
+                response = rest.postForEntity(api, request, String.class, queries);
+            } else {
+                response = rest.postForEntity(api, request, String.class);
+            }
+            return response.getBody();
+        } catch (Exception e) {
+            System.out.println("## Error :: "+e.getMessage());
```

Comment:
Please log the exception. Do not print it.
Log it like this:
logger.error("Some message..." + e);